Added use of types in node - issue#2149 (#2218)
nodejs/nodejs.dev: src/documentation/0056-node-with-typescript/index.md

Some of the other benefits of TypeScript that are worth mentioning are that it can be adopted progressively, it helps making code more readable and understandable and it allows developers to use modern language features while shipping code for older Node.js versions.

## Using node packages with TypeScript

> Node packages generally are written in javascript and not TypeScript so we need to have type definitions for the packages.

Reviewer: ```suggestion
Node packages generally are written in JavaScript and not TypeScript so we need to have type definitions for the packages.
```

> For this we need to install a third party package called `@types/node`.

Reviewer 2: ```suggestion
To use the built-in node modules like `http/stream` with Typescript, we need to install the type definitions for them.
```

```js
npm install -D @types/node
```

> 

Reviewer 2: ```js
import http, { RequestOptions, IncomingMessage } from "http";

const options: RequestOptions = {
  path: "/api/test",
  headers: {
    "Content-Type": "application/json",
  },
  host: "localhost",
};

const handler = async (res: IncomingMessage) => {
  const data: any[] = [];
  res.on("data", (chunk) => {
    data.push(chunk);
  });
  res.on("end", () => {
    console.log(data.toString());
  });
};

http.request(options, handler).end();
```

> For example

Reviewer 2: ```suggestion
We can also install type definitions for third party modules if they don't have the type definitions in themselves
```

first we install a node pacakge (say express)

```js
npm install express
```

> Then we will install the type definations for express

Reviewer: ```suggestion
Then we will install the type definitions for express
```

```js
npm install -D @types/express
```

> require the package and its type definations

Reviewer: ```suggestion
require the package and its type definitions
```

```js
import express, {Request,Response,Application} from 'express';
```

> initalize express with typeScript

Reviewer: ```suggestion
initialize express with TypeScript
```

```js
const app:Application = express();
```

> setting an endpoint in

Reviewer: ```suggestion
Create a root endpoint
```

> ```js

Reviewer 2: I think this part can be merged with the codeblock before it

Some of the other benefits of TypeScript that are worth mentioning are that it can be adopted progressively, it helps making code more readable and understandable and it allows developers to use modern language features while shipping code for older Node.js versions.

## Using node packages with TypeScript

Node packages generally are written in Javascript and not TypeScript so we need to have type definitions for the packages.
To use the built-in node modules (eg. 'http/stream') we need to install the type definitions

```js
npm install -D @types/node
```

To use the built-in node module with TypeScript

```js
import http, { RequestOptions, IncomingMessage } from "http";

const options: RequestOptions = {
  path: "/api/test",
  headers: {
    "Content-Type": "application/json",
  },
  host: "localhost",
};

const handler = async (res: IncomingMessage) => {
  const data: any[] = [];
  res.on("data", (chunk) => {
    data.push(chunk);
  });
  res.on("end", () => {
    console.log(data.toString());
  });
};

http.request(options, handler).end();
```

> We can also install type definitions for third party modules if they don't have the type definations in themselves

Reviewer: ```suggestion
We can also install type definitions for third party modules if they don't have the type definitions in themselves
```

Such as

> ```js

Reviewer: ```suggestion
```bash
```

npm install express
```

Then we will install the type definitions for express

> ```js

Reviewer: ```suggestion
```bash
```
